src/Array.ts

```diff
@@ -741,7 +742,21 @@ export const rotate = <A>(n: number, xs: Array<A>): Array<A> => {
  */
 export const uniq = <A>(S: Setoid<A>) => (as: Array<A>): Array<A> => {
   const r: Array<A> = []
-  new Set(as).forEach(a => r.push(a))
+  const set: Set<A> = new Set()
+  const isMember = member(S)(set)
+  const l = as.length
+  let i = 0
+  while (i < l) {
```

Comment:
If I remember well, while is slower that for loops, hence the usage of for loop in the Array.ts file when we were optimising for speed. 
(not a proof: https://cdn-images-1.medium.com/max/2000/1*0b2Yf4R7eiETephfEjE2HA.png)

Reply:
@sledorze did you see my proposal (i.e. not using `Set`)? WDYT?

Reply:
@gcanti I'm in favor of not using Set in this use case, too, to keep things lean.